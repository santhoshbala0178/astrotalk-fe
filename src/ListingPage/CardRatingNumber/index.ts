import CardRatingNumber from './CardRatingNumber';

export default CardRatingNumber;
