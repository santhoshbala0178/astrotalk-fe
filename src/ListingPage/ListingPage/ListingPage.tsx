import { gql, useQuery, useReactiveVar } from '@apollo/client';
import React, { useState } from 'react';
import styled from 'styled-components';
import { searchNameVar, sortByVar } from '../../cache';
import Loader from '../../Loader';
import Card from '../Card';
import LoadMore from '../LoadMore';
import SubHeader from '../SubHeader';

const StyledListingPage = styled.div`
  margin-left: 12%;
`;

const StyledCardHolder = styled.div`
  margin: 25px;
  display: flex;
  flex-wrap: wrap;
`;

export const ASTROLOGER_TILE_DATA = gql`
  fragment AstrologerData on Astrologer {
    id
    name
    experience
    languages
    specialities
    rates {
      chatRate
      callRate
      reportRate
    }
    reviewDetails {
      overallRating
      totalReviews
      individualRating1: individualRating(rating: RATING_1)
      individualRating2: individualRating(rating: RATING_2)
      individualRating3: individualRating(rating: RATING_3)
      individualRating4: individualRating(rating: RATING_4)
      individualRating5: individualRating(rating: RATING_5)
    }
  }
`;

export const GET_ASTROLOGERS = gql`
  query GetAstrologers($sortBy: SortEnum, $searchName: String, $after: String) {
    astrologers(sortBy: $sortBy, searchName: $searchName, after: $after) {
      cursor,
      hasMore,
      astrologers {
        ...AstrologerData
      }
    }
  }
  ${ASTROLOGER_TILE_DATA}
`;

const ListingPage: React.FC = () => {
  const sortBy = useReactiveVar(sortByVar);
  const searchName = useReactiveVar(searchNameVar);
  const [isLoadingMore, changeIsLoadingMore] = useState(false);

  const { data, loading, error, fetchMore} = useQuery(GET_ASTROLOGERS, {
    variables: { sortBy, searchName},
  });


  async function loadMore() {
    changeIsLoadingMore(true)
    await fetchMore({
      variables: {
        sortBy, searchName,
        after: data.astrologers.cursor
      }
    });
    changeIsLoadingMore(false)
  }

  return (
    <StyledListingPage>
      <SubHeader />
      <StyledCardHolder>
        {loading && <Loader />}
        {error && <div>Error in Loading listing page</div>}
        {data &&
          data.astrologers.astrologers.map((astrologer, index) => (
            <Card astrologer={astrologer} key={`${index + 1}`} />
          ))}
      </StyledCardHolder>
      {data && data.astrologers &&
        data.astrologers.hasMore &&
        (
          isLoadingMore?<Loader/>: <LoadMore loadMore={loadMore} />
        )}
    </StyledListingPage>
  );
};

export default ListingPage;
