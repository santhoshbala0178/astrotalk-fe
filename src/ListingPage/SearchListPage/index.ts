import SearchListPage from './SearchListPage';

export default SearchListPage;
