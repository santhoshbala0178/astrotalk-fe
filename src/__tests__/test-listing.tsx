import React from 'react';
import { InMemoryCache } from '@apollo/client';
import '@testing-library/jest-dom';
import ListingPage, { GET_ASTROLOGERS } from '../ListingPage/ListingPage/ListingPage';
import {
  renderApollo,
  cleanup,
  screen
} from '../test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import { fireEvent } from '@testing-library/react';


const mockAstrologers = [
  {
    __typename: 'Astrologer',
    id: '1',
    name: 'Bharath',
    experience: 7,
    languages: ['English', 'Hindi'],
    specialities: ['Tarot', 'Numerology'],
    aboutSelf:
      'Bharath is a renowned Astrologer from India with an experience of more than 07 years in the field of Astrology. He is an expert Vedic Astrologer. He is an IIT graduate and his belief in ancient Hindu techniques and astrology gave him the confidence to reach out to the larger public and help them overcome their problems in their lives with the help of astrology. He believes in providing the most productive and soothing remedies to our customers. His area of expertise Marriage problems, Property, Health issues, and Vastu.',
    rates: {
      __typename: 'Rates',
      chatRate: 7,
      callRate: 10,
      reportRate: 110,
    },
    work: {
      __typename: 'Work',
      chatMinutes: '8K',
      callMinutes: '10K',
      reports: 12,
    },
    reviewDetails: {
      __typename: 'ReviewDetails',
      overallRating: 4.86,
      totalReviews: 1829,
      individualRating5: 1219,
      individualRating4: 200,
      individualRating3: 250,
      individualRating2: 100,
      individualRating1: 50,
    },
    allReviews: [
      {
        __typename: 'Review',
        rating: 4,
        date: 1614959127,
        reviewText: 'thanks for the help!!',
      },
      {
        __typename: 'Review',
        rating: 2,
        date: 1514959127,
        reviewText: 'this is outrageous',
      },
    ],
  },
  {
    __typename: 'Astrologer',
    id: '2',
    name: 'Pooja',
    experience: 4,
    languages: ['English', 'Hindi'],
    specialities: ['Tarot', 'Numerology'],
    aboutSelf:
      'Bharath is a renowned Astrologer from India with an experience of more than 07 years in the field of Astrology. He is an expert Vedic Astrologer. He is an IIT graduate and his belief in ancient Hindu techniques and astrology gave him the confidence to reach out to the larger public and help them overcome their problems in their lives with the help of astrology. He believes in providing the most productive and soothing remedies to our customers. His area of expertise Marriage problems, Property, Health issues, and Vastu.',
    rates: {
      __typename: 'Rates',
      chatRate: 2,
      callRate: 10,
      reportRate: 110,
    },
    work: {
      __typename: 'Work',
      chatMinutes: '8K',
      callMinutes: '10K',
      reports: 12,
    },
    reviewDetails: {
      __typename: 'ReviewDetails',
      overallRating: 2.86,
      totalReviews: 2500,
      individualRating5: 1219,
      individualRating4: 200,
      individualRating3: 250,
      individualRating2: 100,
      individualRating1: 50,
    },
    allReviews: [
      {
        __typename: 'Review',
        rating: 4,
        date: 1614959127,
        reviewText: 'thanks for the help!!',
      },
      {
        __typename: 'Review',
        rating: 2,
        date: 1514959127,
        reviewText: 'this is outrageous',
      },
    ],
  },
  {
    __typename: 'Astrologer',
    id: '3',
    name: 'Sandy',
    experience: 7,
    languages: ['English', 'Hindi'],
    specialities: ['Tarot', 'Numerology'],
    aboutSelf:
      'Bharath is a renowned Astrologer from India with an experience of more than 07 years in the field of Astrology. He is an expert Vedic Astrologer. He is an IIT graduate and his belief in ancient Hindu techniques and astrology gave him the confidence to reach out to the larger public and help them overcome their problems in their lives with the help of astrology. He believes in providing the most productive and soothing remedies to our customers. His area of expertise Marriage problems, Property, Health issues, and Vastu.',
    rates: {
      __typename: 'Rates',
      chatRate: 12,
      callRate: 10,
      reportRate: 110,
    },
    work: {
      __typename: 'Work',
      chatMinutes: '8K',
      callMinutes: '10K',
      reports: 12,
    },
    reviewDetails: {
      __typename: 'ReviewDetails',
      overallRating: 1,
      totalReviews: 1029,
      individualRating5: 1219,
      individualRating4: 200,
      individualRating3: 250,
      individualRating2: 100,
      individualRating1: 50,
    },
    allReviews: [
      {
        __typename: 'Review',
        rating: 4,
        date: 1614959127,
        reviewText: 'thanks for the help!!',
      },
      {
        __typename: 'Review',
        rating: 2,
        date: 1514959127,
        reviewText: 'this is outrageous',
      },
    ],
  },
];

describe('Listing Page', () => {
  afterEach(cleanup);

  it('Snapshot Test Listing  Page', async () => {
    const cache = new InMemoryCache({ addTypename: false });
    const mocks = [
      {
        request: {
          query: GET_ASTROLOGERS,
          variables: { sortBy: 'NO_SORT', searchName: '' },
        },
        result: {
          data: {
            astrologers: {
              cursor: '3',
              hasMore: true,
              astrologers: mockAstrologers,
            },
          },
        },
      },
    ];

    const { container, debug, getByText, findAllByText } = await renderApollo(
      <Router>
        <ListingPage />
    </Router>,
      {
        mocks,
        cache,
      }
    );
    await findAllByText('Bharath');

    expect(container).toMatchSnapshot();
  });

  it('render Listing  Page', async () => {
    const cache = new InMemoryCache({ addTypename: false });
    const mocks = [
      {
        request: {
          query: GET_ASTROLOGERS,
          variables: { sortBy: 'NO_SORT', searchName: '' },
        },
        result: {
          data: {
            astrologers: {
              cursor: '3',
              hasMore: true,
              astrologers: mockAstrologers,
            },
          },
        },
      },
    ];

    const { container, debug, getByText, findAllByText } = await renderApollo(
      <Router>
        <ListingPage />
    </Router>,
      {
        mocks,
        cache,
      }
    );
    await findAllByText('Bharath');

    expect(getByText('Bharath')).toBeInTheDocument();
  });

  it('test count of cards', async () => {
    const cache = new InMemoryCache({ addTypename: false });
    const mocks = [
      {
        request: {
          query: GET_ASTROLOGERS,
          variables: { sortBy: 'NO_SORT', searchName: '' },
        },
        result: {
          data: {
            astrologers: {
              cursor: '3',
              hasMore: true,
              astrologers: mockAstrologers,
            },
          },
        },
      },
    ];

    const { container, debug, getAllByTestId, findAllByText } = await renderApollo(
      <Router>
        <ListingPage />
    </Router>,
      {
        mocks,
        cache,
      }
    );
    await findAllByText('Bharath');

    expect(getAllByTestId('astro-card')).toHaveLength(3);
  });

  it('test filter of name', async () => {
    const cache = new InMemoryCache({ addTypename: false });
    const mocks = [
      {
        request: {
          query: GET_ASTROLOGERS,
          variables: { sortBy: 'NO_SORT', searchName: '' },
        },
        result: {
          data: {
            astrologers: {
              cursor: '3',
              hasMore: true,
              astrologers: mockAstrologers,
            },
          },
        },
      },
    ];

    const {
      container,
      debug,
      getByTestId,
      getAllByTestId,
      findByText,
    } = await renderApollo(
      <Router>
        <ListingPage />
    </Router>,
      {
        mocks,
        cache,
      }
    );
    await findByText('Bharath');

    const inputElement = getByTestId('search-input');
    fireEvent.change(inputElement, { target: { value: 'poo' } });

    expect(inputElement.textContent).toBe('poo');
  });
  it('test count of cards', async () => {
    const cache = new InMemoryCache({ addTypename: false });
    const mocks = [
      {
        request: {
          query: GET_ASTROLOGERS,
          variables: { sortBy: 'NO_SORT', searchName: '' },
        },
        result: {
          data: {
            astrologers: {
              cursor: '3',
              hasMore: true,
              astrologers: mockAstrologers,
            },
          },
        },
      },
    ];

    const { container, debug, getAllByTestId, findAllByText } = await renderApollo(
      <Router>
        <ListingPage />
    </Router>,
      {
        mocks,
        cache,
      }
    );
    await findAllByText('Bharath');

    expect(getAllByTestId('astro-card')).toHaveLength(3);
  });
});
