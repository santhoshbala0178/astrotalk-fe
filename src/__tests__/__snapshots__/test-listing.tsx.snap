// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing Page Snapshot Test Listing  Page 1`] = `
<div>
  <div
    class="sc-bBXqnf kUNHDl"
  >
    <div
      class="sc-hHftDr kagUfo"
    >
      <div
        class="sc-dmlrTW jHepxZ"
      >
        <div
          class="sc-fKFyDc bdRfkw"
        >
          Chat with Astrologer
        </div>
      </div>
      <div
        class="sc-kfzAmx eruYhV"
      >
        <div
          class="sc-idOhPF jEdYuu"
        >
          <select
            class="sc-dIUggk dHjHTc"
          >
            <option
              value="NO_SORT"
            >
              Sort By
            </option>
            <option
              value="EXP_HIGH"
            >
              Experience: High to Low
            </option>
            <option
              value="EXP_LOW"
            >
              Experience: Low to High
            </option>
            <option
              value="PRICE_HIGH"
            >
              Price: High to Low
            </option>
            <option
              value="PRICE_LOW"
            >
              Price: Low to High
            </option>
            <option
              value="RATING_HIGH"
            >
              Rating: High to Low
            </option>
          </select>
        </div>
        <div
          class="sc-fFubgz jiNEGY"
        >
          <input
            class="sc-bkzZxe eqYwcg"
            data-testid="search-input"
            placeholder="Search name...."
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="sc-iwyYcG ggyuIB"
    >
      <div
        class="sc-iqHYGH fiQSDt"
        data-testid="astro-card"
      >
        <div
          class="sc-crrsfI GNQKu"
        >
          <div
            class="sc-dlfnbm brtggE"
          >
            <a
              class="sc-eCssSg glAyEi"
              href="/best-astrologer/bharath"
            >
              <img
                class="sc-hKgILt fMBXVU"
                height="40px"
                src="/assets/TAT_web_logo2.png"
                width="40px"
              />
            </a>
          </div>
          <div
            class="sc-dQppl hMnvEw"
          >
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
          </div>
          <div
            class="sc-pFZIQ jdFpGQ"
          >
            <img
              alt="Total Rating"
              class="sc-kEjbxe iwwVPE"
              height="8px"
              src="/assets/rating.png"
              width="8px"
            />
            <span
              class="sc-jrAGrp cApaSj"
            >
              1829 total
            </span>
          </div>
        </div>
        <div
          class="sc-bqyKva jmnZnB"
        >
          <div
            class="sc-jSgupP jwkMIT"
          >
            <a
              class="sc-gKsewC VvBIK"
              href="/best-astrologer/1"
            >
              Bharath
            </a>
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            English, Hindi
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Tarot, Numerology
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Exp: 7 Years
          </div>
          <div>
            <img
              alt="Rupee"
              class="sc-fubCfw duHFTV"
              height="10px"
              src="/assets/rupee.svg"
              width="10px"
            />
            <span
              class="sc-iBPRYJ hasEMx"
            >
              7
              /min
            </span>
          </div>
        </div>
        <div
          class="sc-kstrdz glrsGB"
        />
      </div>
      <div
        class="sc-iqHYGH fiQSDt"
        data-testid="astro-card"
      >
        <div
          class="sc-crrsfI GNQKu"
        >
          <div
            class="sc-dlfnbm brtggE"
          >
            <a
              class="sc-eCssSg glAyEi"
              href="/best-astrologer/bharath"
            >
              <img
                class="sc-hKgILt fMBXVU"
                height="40px"
                src="/assets/TAT_web_logo2.png"
                width="40px"
              />
            </a>
          </div>
          <div
            class="sc-dQppl hMnvEw"
          >
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
          </div>
          <div
            class="sc-pFZIQ jdFpGQ"
          >
            <img
              alt="Total Rating"
              class="sc-kEjbxe iwwVPE"
              height="8px"
              src="/assets/rating.png"
              width="8px"
            />
            <span
              class="sc-jrAGrp cApaSj"
            >
              2500 total
            </span>
          </div>
        </div>
        <div
          class="sc-bqyKva jmnZnB"
        >
          <div
            class="sc-jSgupP jwkMIT"
          >
            <a
              class="sc-gKsewC VvBIK"
              href="/best-astrologer/2"
            >
              Pooja
            </a>
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            English, Hindi
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Tarot, Numerology
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Exp: 4 Years
          </div>
          <div>
            <img
              alt="Rupee"
              class="sc-fubCfw duHFTV"
              height="10px"
              src="/assets/rupee.svg"
              width="10px"
            />
            <span
              class="sc-iBPRYJ hasEMx"
            >
              2
              /min
            </span>
          </div>
        </div>
        <div
          class="sc-kstrdz glrsGB"
        />
      </div>
      <div
        class="sc-iqHYGH fiQSDt"
        data-testid="astro-card"
      >
        <div
          class="sc-crrsfI GNQKu"
        >
          <div
            class="sc-dlfnbm brtggE"
          >
            <a
              class="sc-eCssSg glAyEi"
              href="/best-astrologer/bharath"
            >
              <img
                class="sc-hKgILt fMBXVU"
                height="40px"
                src="/assets/TAT_web_logo2.png"
                width="40px"
              />
            </a>
          </div>
          <div
            class="sc-dQppl hMnvEw"
          >
            <img
              alt="Star Rating"
              class="sc-gsTCUz fOtSmi"
              height="15px"
              src="/assets/star.svg"
              width="15px"
            />
          </div>
          <div
            class="sc-pFZIQ jdFpGQ"
          >
            <img
              alt="Total Rating"
              class="sc-kEjbxe iwwVPE"
              height="8px"
              src="/assets/rating.png"
              width="8px"
            />
            <span
              class="sc-jrAGrp cApaSj"
            >
              1029 total
            </span>
          </div>
        </div>
        <div
          class="sc-bqyKva jmnZnB"
        >
          <div
            class="sc-jSgupP jwkMIT"
          >
            <a
              class="sc-gKsewC VvBIK"
              href="/best-astrologer/3"
            >
              Sandy
            </a>
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            English, Hindi
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Tarot, Numerology
          </div>
          <div
            class="sc-bdfBwQ bAedth"
          >
            Exp: 7 Years
          </div>
          <div>
            <img
              alt="Rupee"
              class="sc-fubCfw duHFTV"
              height="10px"
              src="/assets/rupee.svg"
              width="10px"
            />
            <span
              class="sc-iBPRYJ hasEMx"
            >
              12
              /min
            </span>
          </div>
        </div>
        <div
          class="sc-kstrdz glrsGB"
        />
      </div>
    </div>
    <div
      class="sc-hBEYos hrYMlc"
    >
      <button
        class="sc-fodVxV fzKvnv"
        data-testid="load-more"
      >
        Load more...
      </button>
    </div>
  </div>
</div>
`;
